jcl/src/java.base/share/classes/openj9/internal/criu/InternalCRIUSupport.java

```diff
@@ -972,6 +974,42 @@ public synchronized void checkpointJVM() {
 				J9InternalCheckpointHookAPI.registerPostRestoreHook(HookMode.SINGLE_THREAD_MODE, RESTORE_CLEAR_INETADDRESS_CACHE_PRIORITY, "Clear InetAddress cache on restore", InternalCRIUSupport::clearInetAddressCache); //$NON-NLS-1$
 				J9InternalCheckpointHookAPI.registerPostRestoreHook(HookMode.SINGLE_THREAD_MODE, RESTORE_ENVIRONMENT_VARIABLES_PRIORITY, "Restore system properties", InternalCRIUSupport::setRestoreJavaProperties); //$NON-NLS-1$
 
+				/* Add option overrides */
+				Properties props = VM.internalGetProperties();
+
+				String unprivilegedOpt = props.getProperty("openj9.internal.criu.unprivilegedMode"); //$NON-NLS-1$
+				if (unprivilegedOpt != null) {
+						setUnprivileged(Boolean.parseBoolean(unprivilegedOpt));
+				}
+
+				String tcpEstablishedOpt = props.getProperty("openj9.internal.criu.tcpEstablished"); //$NON-NLS-1$
+				if (tcpEstablishedOpt != null) {
+					setTCPEstablished(Boolean.parseBoolean(tcpEstablishedOpt));
+				}
+
+				String ghostFileLimitOpt = props.getProperty("openj9.internal.criu.ghostFileLimit"); //$NON-NLS-1$
+				if (ghostFileLimitOpt != null) {
+					try {
+						setGhostFileLimit(Long.parseLong(ghostFileLimitOpt));
+					} catch (NumberFormatException e) {
+						System.err.println("Invalid value for `-Dopenj9.internal.criu.ghostFileLimit` specified.");
+					}
+				}
+
+				String logLevelOpt =  props.getProperty("openj9.internal.criu.logLevel"); //$NON-NLS-1$
+				if (logLevelOpt != null) {
+					try {
+						setLogLevel(Integer.parseInt(logLevelOpt));
+					} catch (NumberFormatException e) {
+						System.err.println("Invalid value for `-Dopenj9.internal.criu.logLevel` specified.");
```

Comment:
Also suggest printing the system property value supplied, and adding `//$NON-NLS-1$`.